homeassistant/components/netgear/sensor.py

```diff
@@ -306,14 +350,22 @@ def __init__(
         self._name = f"{router.device_name} {entity_description.name}"
         self._unique_id = f"{router.serial_number}-{entity_description.key}-{entity_description.index}"
 
-        self._value = None
+        self._value: str | None = None
         self.async_update_device()
 
     @property
     def native_value(self):
         """Return the state of the sensor."""
         return self._value
 
+    async def async_added_to_hass(self) -> None:
+        """Handle entity which will be added."""
+        await super().async_added_to_hass()
+        if self.coordinator.data is None:
+            state = await self.async_get_last_state()
+            if state is not None:
```

Comment:
I think it is fine to restore all of them, it does not matter for all other sensors since they get updated while loading the integration anyway.

A speed sensor update will test the full bandwidth of your ethernet connection directly from the router. Therefore this will take up all available bandwidth on your ethernet connection for about 15 to 30 seconds. I think it is very inconveniant to do this at HomeAssistant start since all other integrations might need the ethernet connection, so it is not nice to max it out during start for 15 to 30 seconds. That is why I chose to only do the first update after 30 minutes.

That is also why the scan interval is 30 minutes, you don't want to max out your ethernet connection all the time.

But it is anoying in the graphs if there is a gap of 30 minutes because the sensor is unavailable at start, so I just restore the previous result.